v2/devdoc/logging.md

```diff
@@ -0,0 +1,339 @@
+    LOG_CONTEXT_DESTROY(dynamically_allocated_log_context);
+```
+
+### Context chaining
+
+It shall be supported to chain contexts (define a context or create a context dynamically while specifying a parent context to inherit the information from).
+
+### LOG_CONTEXT_PROPERTY
+
+`LOG_CONTEXT_PROPERTY` is a macro that allows defining non-string properties by specifying a property type, name and a value.
+
+```c
+#define LOG_CONTEXT_PROPERTY(property_type, property_name, property_value) \
+    ...
+```
+
+The following types shall be supported out of the box:
+
+- ansi_charptr
+- wchar_t_ptr
+- int8_t
+- uint8_t
+- int16_t
+- uint16_t
+- int32_t
+- uint32_t
+- int64_t
+- uint64_t
+
+Note: more types will be added as needed.
+
+### LOG_CONTEXT_STRING_PROPERTY
+
+`LOG_CONTEXT_STRING_PROPERTY` is a macro that allows defining context properties by formatting the value as a string with `printf`-like formatting.
+
+```c
+#define LOG_CONTEXT_STRING_PROPERTY(property_name, property_format_specifier, property_value) \
+    ...
+```
+
+### LOG_CONTEXT_NAME
+
+```
+#define LOG_CONTEXT_NAME(context_name) \
+```
+
+`LOG_CONTEXT_NAME` shall allow specifying the context name to be used either when printing logs to console or when ETW events are published.
+
+By default if not specified, the context name shall be `unnamed`.
+
+Example:
+
+```c
+    LOG_CONTEXT_HANDLE context_1;
+    LOG_CONTEXT_CREATE(context_1, NULL, LOG_CONTEXT_NAME(xxx), LOG_CONTEXT_STRING_PROPERTY(name, "%s", "haga"));
+
+    LOGGER_LOG(LOG_LEVEL_ERROR, context_1, "log error with custom context name");
+```
+
+This shall output:
+
+```
+LOG_LEVEL_ERROR Time:Mon Aug 15 11:16:09 2022 File:G:\w\c-logging\v2\tests\logger_int\main.c:102 Func:main xxx={ name=haga  }  log error with custom context name
+```
+
+### Extensions
+
+An extensions set of properties shall be available in order to ease writing logging code.
+Examples of what would fall in the extensions category are:
+
+- syntactic sugar for a message property
+- `GetLastError` on Windows
+- `HRESULT` on Windows
+
+### LOG_MESSAGE
+
+`LOG_MESSAGE` is a macro that is syntactic sugar to wrap `LOG_CONTEXT_STRING_PROPERTY` in order to output a `message` property, while typing less.
+
+Example:
+
+```c
+LOGGER_LOG_EX(LOG_LEVEL_ERROR, 
+    LOG_CONTEXT_STRING_PROPERTY(int32_t, an_integer, 42),
+    LOG_MESSAGE("hello world"));
+```
+
+## LOGGER_LOG_EX
+
+```c
+#define LOGGER_LOG_EX(log_level, ...) \
+    ...
+```
+
+`LOGGER_LOG_EX` allows logging directly with structured data without explicitly creating a context.
+
+`LOGGER_LOG_EX` creates a stack context under the hood and logs the data of that context.
+
+Example:
+
+```c
+LOGGER_LOG_EX(LOG_LEVEL_ERROR, 
+    LOG_CONTEXT_STRING_PROPERTY(name, "%s%s", "go", "gu"),
+    LOG_CONTEXT_PROPERTY(int32_t, age, 42));
+```
+
+This would output:
+
+```
+LOG_LEVEL_ERROR Time:Mon Aug 15 11:15:35 2022 File:G:\w\c-logging\v2\tests\logger_int\main.c:125 Func:main unnamed={ name=gogu age=42  }
+```
```

Comment:
I will follow this PR once I will be able to do so again. Meanwhile, signing off.